src/Guide/Api/Utils.hs

```diff
@@ -48,6 +48,8 @@ newtype (?) (field :: *) (help :: Symbol) = H field
 instance ToJSON field => ToJSON (field ? help) where
   toJSON (H a) = toJSON a
 
+instance FromJSON field => FromJSON (field ? help)
```

Comment:
This instance should be written explicitly (like `ToJSON`), because otherwise it uses a generic instance and AFAIK wraps the value into something.

Generally, whenever you have two symmetric things (in this case ToJSON and FromJSON) and you use different approaches for them, it might be a sign that something is wrong here. And even if nothing is wrong, being consistent still helps readability.